### PR TITLE
Fix blurry DefaultTtfFontAsBitmap text in CachedContainer when virtual size < window size

### PR DESCRIPTION
**Pull request by eaboll**
Adds `CachedContainer.expensiveScaling: Boolean?`, which enables scaling of the content based on the actual window size instead of just the virtual size. Set to `true` by default. 

This fixes part of issue #1748. The default non-bitmap font is still blurry regardless of whether or not it's in a CachedContainer.

In the sandbox:

image

### CODE REVIEW

**Comment by soywiz** [comment]
Awesome. Thanks for implementing it!

I have noticed a small tweak that could improve it further if you can change it

File: korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt [hunk 1]
```diff
@@ -36,7 +36,9 @@ open class FixedSizeCachedContainer(
 
 open class CachedContainer(
     @property:ViewProperty
-    var cache: Boolean = true
+    var cache: Boolean = true,
+    @property:ViewProperty
+    var expensiveScaling: Boolean = true,
```
Comment (soywiz) [inline korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt:41]: Can we have `expensiveScaling: Boolean? = null` ?

When constructing KorGE we have a `quality` parameter. Can we use the quality parameter to determine the value when not specified?

Here's the quality:
https://github.com/korlibs/korge/blob/b9213f1adc4d66c2148fca9363ca520874a1a696/korgw/src/commonMain/kotlin/korlibs/render/GameWindow.kt#L480

Comment (soywiz) [reply korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt:41]: This PR https://github.com/korlibs/korge/pull/1762/files will be needed, so it is possible to access tue GameWindow quality inside the RenderContext

Comment (eaboll) [reply korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt]: > Can we have `expensiveScaling: Boolean? = null` ?

Excellent idea, implemented! https://github.com/korlibs/korge/pull/1760/commits/59cb48385fc87c92190146f4d759a8187d3cdd5e

File: korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt [hunk 2]
```diff
@@ -84,16 +84,21 @@ open class CachedContainer(
...
+        val renderScale: Float = when (ctx.quality) {
             GameWindow.Quality.PERFORMANCE -> 1f
             else -> ctx.devicePixelRatio
         }
         //val renderScale = 1.0
 
+        val doExpensiveScaling = expensiveScaling ?: when(ctx.quality) {
+            GameWindow.Quality.PERFORMANCE -> false
+            else -> true
+        }
```
Comment (soywiz) [inline korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt:96]: Do you think that by adding some properties the Quality enum could we avid somehow the when so the code is simpler?

File: korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt [hunk 3]
```diff
@@ -36,7 +36,9 @@ open class FixedSizeCachedContainer(
 
 open class CachedContainer(
     @property:ViewProperty
-    var cache: Boolean = true
+    var cache: Boolean = true,
+    @property:ViewProperty
+    var expensiveScaling: Boolean? = null,
```
Comment (soywiz) [inline korge/src/commonMain/kotlin/korlibs/korge/view/CachedContainer.kt:41]: I'm not sure about this name. Could we pass a `quality: Quality` enum, or rename it? Like `hidpiAware: Boolean?` or other names? People might ask: In which way is this expensive? What scaling mean in this context?

**Comment by soywiz** [comment]
Thanks @eaboll for implementing this!

To avoid this getting outdated, I'm going to merge it and make a few adjustments.
I'm going to change `CachedContainer.expensiveScaling` to `CachedContainer.quality` and going to adjust Quality to be accessible from outside, and to simplify the when.

**Pull request merged**